### ISSUE TITLE
Crash with Demo app while panning and zooming with latest snapshot

### ISSUE THREAD

**Issue by mkv123** [mapbox/mapbox-gl-native#6370]
**Platform:** Android
**Mapbox SDK version:** Latest snapshot

### Steps to trigger behavior

 1. Start Simple Map demo
 2. Zoom and pan
 3.

### Expected behavior
Snappy wonderful map viewing.

### Actual behavior
Crash.

[mapbox_crash.txt](https://github.com/mapbox/mapbox-gl-native/files/480017/mapbox_crash.txt)

Reproducable every time on this Samsung Galaxy Note II with Android 4.4. Originally noticed in own app, but also easily replicated on demo app. 

In ```A/libc: Fatal signal 11 (SIGSEGV) at 0x00000030 (code=1), thread 1165 (pboxandroiddemo) ``` the SIGSEV at 0x00000030 is always the same (thread number varies of course)

**Comment by mkv123** [comment]
Also reproduced on a Samsung Galaxy S4 with Android 5.0.1.

```09-19 16:09:58.056 11732-11732/com.mapbox.mapboxandroiddemo A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x30 in tid 11732 (pboxandroiddemo)```

[mapbox_crash_s4.txt](https://github.com/mapbox/mapbox-gl-native/files/480042/mapbox_crash_s4.txt)

**Comment by mkv123** [comment]
Logs always show ```[JNI]: nativeQueryPointAnnotations``` as last print before crash.

**Comment by mkv123** [comment]
Also reproduced on Nexus 6P running Android 6. I haven't come across a device where this wouldn't happen, and the address is always 0x30 and the last log line is always ```com.mapbox.mapboxandroiddemo D/mbgl: {pboxandroiddemo}[JNI]: nativeQueryPointAnnotations```

Crash log:
[mapbox_crash_nexus_6p.txt](https://github.com/mapbox/mapbox-gl-native/files/480113/mapbox_crash_nexus_6p.txt)

**Comment by tobrun** [comment]
@mkv123 I'm not seeing this with the latest master, which is equivalent to latests snapshot. Could you share what you are doing with the map? Adding markers etc?

**Comment by mkv123** [comment]
@tobrun: Not doing anything special, just opening the simple demo and zooming out a bunch of times, then zooming in. Typically with maybe 1s break between gestures. In 30s I have a crash. Exactly the same usage pattern as #6374 so it might be related, although the crash is with a different address and a different signal.

**Comment by jfirebaugh** [comment]
Based on `[JNI]: nativeQueryPointAnnotations`, I'm pretty sure this is a duplicate of #6374.

**Comment by jfirebaugh** [comment]
Let me know if you see this after #6376 (tonight's snapshot).

**Comment by mkv123** [comment]
I can confirm I'm no longer seeing this issue.
